tests/test_runner/tests.py

```diff
@@ -734,3 +736,16 @@ def test_run_checks_passes_and_teardown_raises(self):
                     runner.run_tests(['test_runner_apps.sample.tests_sample.TestDjangoTestCase'])
             self.assertTrue(teardown_databases.called)
             self.assertFalse(teardown_test_environment.called)
+
+
+class DiscoverRunnerDeprecationTests(SimpleTestCase):
+    def test_extra_tests_deprecated(self):
+        runner = NoOpTestRunner(verbosity=0, interactive=False)
+        runner.stream = StringIO()
```

Comment:
Maybe just do this in the `__init__()` method of your subclass, after calling `super()`? That will keep all your modifications together in one place.